src/Provider/ArrayProvider.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of Swap.
+ *
+ * (c) Florian Voutzinos <florian@voutzinos.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swap\Provider;
+
+use Swap\Exception\UnsupportedCurrencyPairException;
+use Swap\Model\CurrencyPair;
+use Swap\ProviderInterface;
+
+/**
+ * Provides rates from an array.
+ *
+ * @author Florian Voutzinos <florian@voutzinos.com>
+ */
+final class ArrayProvider implements ProviderInterface
+{
+    private $rates;
+
+    /**
+     * Constructor.
+     *
+     * @param \Swap\Model\Rate[] $rates An array of rates indexed by the corresponding currency pair symbol
```

Comment:
I think storing instances of Rate objects is not the best idea. Array providers are possible created from files, looping through each entry and creating a rate object can cost resources. I think a simple associative array is more than enough. The index is the currency pair, the element is either an array with rate and datetime or just the rate.